client/src/components/Feed/PostSocial.js

```diff
@@ -126,14 +126,44 @@ const PostSocial = ({
           {renderLabels("Like", numLikes)}
         </div>
       ) : (
-        <div
-          id={gtmTag("like", GTM.feed.prefix)}
-          className="social-icon"
-          onClick={() => handlePostLike(id, liked, true)}
-        >
-          {renderLikeIcon()}
-          {renderLabels("Like", numLikes)}
-        </div>
+        <>
+          {isAuthenticated ? (
+            <Link
+              to={{
+                pathname: `/post/${id}`,
+                state: {
+                  postId: id,
+                  comments: true,
+                  from: window.location.href,
+                },
+              }}
+            >
+              <div
+                id={gtmTag("like", GTM.feed.prefix)}
+                className="social-icon"
+                onClick={() => handlePostLike(id, liked, true)}
+              >
+                {renderLikeIcon()}
+                {renderLabels("Like", numLikes)}
+              </div>
+            </Link>
```

Comment:
authenticated user should be able to like directly without the need to redirect them to the post page.

```suggestion
            <div
              id={gtmTag("like", GTM.feed.prefix)}
              className="social-icon"
              onClick={() => handlePostLike(id, liked, true)}
            >
              {renderLikeIcon()}
              {renderLabels("Like", numLikes)}
            </div>
```

Reply:
Good catch thanks @Zaydme